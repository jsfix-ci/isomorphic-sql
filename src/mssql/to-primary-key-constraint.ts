export function toPrimaryKeyConstraint(table: string, columns: string[]) {
    return `ALTER TABLE ${table} ADD CONSTRAINT PK_${table}_${columns.join('_')} 
PRIMARY KEY (${columns.join(', ')})`;
};
