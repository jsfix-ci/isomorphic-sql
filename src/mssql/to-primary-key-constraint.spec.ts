import { toPrimaryKeyConstraint } from './to-primary-key-constraint';

describe('toPrimaryKeyConstraint', () => {
    it('returns an add-constraint expression for a single primary key', () => {
        expect(toPrimaryKeyConstraint('posts', ['slug'])).toMatchSnapshot();
    });

    it('returns an add-constraint expression for a composite primary key', () => {
        expect(toPrimaryKeyConstraint('posts', ['slug', 'author_id'])).toMatchSnapshot();
    });
});