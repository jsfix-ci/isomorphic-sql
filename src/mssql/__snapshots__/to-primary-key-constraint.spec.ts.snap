// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toPrimaryKeyConstraint returns an add-constraint expression for a composite primary key 1`] = `
"ALTER TABLE posts ADD CONSTRAINT PK_posts_slug_author_id 
PRIMARY KEY (slug, author_id)"
`;

exports[`toPrimaryKeyConstraint returns an add-constraint expression for a single primary key 1`] = `
"ALTER TABLE posts ADD CONSTRAINT PK_posts_slug 
PRIMARY KEY (slug)"
`;
